oio/conscience/stats/http.py

```diff
@@ -75,9 +78,13 @@ def get_stats(self):
             # on the remote side but not on the local side, thus we
             # explicitely require the connection to be closed.
             reqid = request_id('stat-')
-            resp = self.agent.pool_manager.request(
-                'GET', self.url, headers={'Connection': 'close',
-                                          REQID_HEADER: reqid})
+            try:
+                with Timeout(self.timeout):
+                    resp = self.agent.pool_manager.request(
+                        'GET', self.url, headers={'Connection': 'close',
+                                                  REQID_HEADER: reqid})
+            except Timeout as e:
+                raise Exception(str(e))
```

Comment:
If one stat fetcher raises an exception, the next stat fetchers won't run. There is something to fix in `oio/conscience/agent.py`.